hbase/src/main/scala/akka/stream/alpakka/hbase/internal/HBaseFlowStage.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.hbase.internal
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.stream.alpakka.hbase.HTableSettings
+import akka.stream.stage._
+import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+import org.apache.hadoop.hbase.client.Table
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ Future, Promise }
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success, Try }
+
+private[hbase] class HBaseFlowStage[A](settings: HTableSettings[A])
+    extends GraphStage[FlowShape[A, Future[A]]]
```

Comment:
This signature is weird, it should be `FlowShape[A, A]` or else you are actually killing backpressure here, as you can emit any number of futures that has not yet completed.